### PR TITLE
doc: update eBPF/compilation instructions v1

### PR DESCRIPTION
**Pull request by lukashino**
Link to Redmine tickets:
- https://redmine.openinfosecfoundation.org/issues/6599
- https://redmine.openinfosecfoundation.org/issues/6686

Describe changes:
- small edit in eBPF instructions to prefer libbpf-dev package to the cloning&compilation
- porting changes from the improved version of installation instructions that got merged into master-6.0.x

### CODE REVIEW

File: doc/userguide/install.rst [hunk 1]
```diff
@@ -140,34 +113,59 @@ one of the following ways::
...
 
 Follow these steps from your Suricata directory::
 
-    ./scripts/bundle.sh
-    ./autogen.sh
     ./configure # you may want to add additional parameters here
     # ./configure --help to get all available parameters
-    make -j8 # j is for paralleling, you may de/increase depending on your CPU
+    # Recommended parameters:
+    CC=clang ./configure --enable-ebpf --enable-ebpf-build --enable-nfqueue \
```
Comment (victorjulien) [inline doc/userguide/install.rst:146]: I don't think we're generally recommending clang over gcc or recommending enabling ebpf and or pretty much any of the other "enable" options. They are disabled by default for a reason

Comment (lukashino) [reply doc/userguide/install.rst:146]: I derived these "enable" options from the packages installed in the previous Recommend dependencies section. As a user, I would be clueless as to why the recommended packages are recommended and what to configure as a next step. This configure step enables most of the "good stuff" that Suricata offers. If you disagree, what would you like me to do? 
1) remove altogether
2) Reword "recommended parameters" to "possible configure parameters"

Comment (lukashino) [reply doc/userguide/install.rst:146]: @victorjulien do you have any preference? I would personally opt for the second option.

Comment (jasonish) [reply doc/userguide/install.rst:146]: I don't think this section should really change.

What would likely be nice is a section on "build options" where these are covered, and what new dependencies they might bring in. A bigger job, but more useful.

Possible example, but a bit verbose: https://nginx.org/en/docs/configure.html

**Comment by lukashino** [comment]
Continues in https://github.com/OISF/suricata/pull/10560

File: doc/userguide/install.rst (hunk 1 above)
Comment (lukashino) [reply doc/userguide/install.rst:146]: Created a ticket: https://redmine.openinfosecfoundation.org/issues/6824

**Pull request closed without merging**